src/index.ts

```diff
@@ -19,24 +19,32 @@ interface InitializeOptions {
   samplingProbability?: number;
   bufferMaxSpans?: number;
   bufferTimeout?: number;
-  ignoreUrls?: RegExp[];
+  ignoreUrls?: (string | RegExp)[];
+  propagateTraceHeaderCorsUrls?: (string | RegExp)[];
 }
 
-const initialize = ({
+export const initializeTracing = ({
   collectionSourceUrl,
   serviceName,
   defaultAttributes,
   samplingProbability = 1,
   bufferMaxSpans = BUFFER_MAX_SPANS,
   bufferTimeout = BUFFER_TIMEOUT,
   ignoreUrls = [],
+  propagateTraceHeaderCorsUrls = [/.*/],
 }: InitializeOptions) => {
-  const providerWithZone = new WebTracerProvider({
+  if (!collectionSourceUrl) {
+    throw new Error(
+      "collectionSourceUrl needs to be defined to initialize SumoLogic OpenTelemetry auto-instrumentation for JavaScript"
+    );
+  }
```

Comment:
Normally these things are better done on type level but this is public API, right? In such a case, it makes sense.

Reply:
This file can be initialized directly using the `<script />` tag and  by providing configuration using `data-` attributes, so typing won't work in such case.